### ISSUE TITLE
[Bug]: Installing Storybook with Sveltekit gives "ERESOLVE overriding peer dependency"

### ISSUE THREAD

**Issue by alexbjorlig** [storybookjs/storybook#25581]
### Describe the bug

When using storybook Sveltekit, `npm i` gives error `ERESOLVE overriding peer dependency`.

[Repo](https://github.com/21RISK/svelte-story-bson) is here.

@JReinhold I tried looking into the package-lock file to better help out, but I got lost 🙃

### To Reproduce

clone and npm i.

### System

```bash
❯ npx storybook@latest info

Storybook Environment Info:

  System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M1 Max
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.17.1 - ~/.nvm/versions/node/v18.17.1/bin/node
    npm: 10.2.5 - ~/.nvm/versions/node/v18.17.1/bin/npm <----- active
    pnpm: 8.12.0 - ~/.nvm/versions/node/v18.17.1/bin/pnpm
  Browsers:
    Chrome: 120.0.6099.216
    Safari: 17.2.1
  npmPackages:
    @storybook/addon-essentials: ^7.6.8 => 7.6.8 
    @storybook/addon-interactions: ^7.6.8 => 7.6.8 
    @storybook/addon-links: ^7.6.8 => 7.6.8 
    @storybook/addon-mdx-gfm: ^7.6.8 => 7.6.8 
    @storybook/blocks: ^7.6.8 => 7.6.8 
    @storybook/svelte: ^7.6.8 => 7.6.8 
    @storybook/sveltekit: ^7.6.8 => 7.6.8 
    @storybook/test: ^7.6.8 => 7.6.8 
    eslint-plugin-storybook: ^0.6.15 => 0.6.15 
    storybook: ^7.6.8 => 7.6.8
```
```


### Additional context

Full error:

```
❯ npm i
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @sveltejs/vite-plugin-svelte@2.5.3
npm WARN Found: vite@5.0.11
npm WARN node_modules/vite
npm WARN   dev vite@"^5.0.3" from the root project
npm WARN   9 more (@storybook/builder-vite, @storybook/svelte-vite, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer vite@"^4.0.0" from @sveltejs/vite-plugin-svelte@2.5.3
npm WARN node_modules/@storybook/svelte-vite/node_modules/@sveltejs/vite-plugin-svelte
npm WARN   @sveltejs/vite-plugin-svelte@"^2.4.2" from @storybook/svelte-vite@7.6.8
npm WARN   node_modules/@storybook/svelte-vite
npm WARN   1 more (@sveltejs/vite-plugin-svelte-inspector)
npm WARN 
npm WARN Conflicting peer dependency: vite@4.5.1
npm WARN node_modules/vite
npm WARN   peer vite@"^4.0.0" from @sveltejs/vite-plugin-svelte@2.5.3
npm WARN   node_modules/@storybook/svelte-vite/node_modules/@sveltejs/vite-plugin-svelte
npm WARN     @sveltejs/vite-plugin-svelte@"^2.4.2" from @storybook/svelte-vite@7.6.8
npm WARN     node_modules/@storybook/svelte-vite
npm WARN     1 more (@sveltejs/vite-plugin-svelte-inspector)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @sveltejs/vite-plugin-svelte-inspector@1.0.4
npm WARN Found: vite@5.0.11
npm WARN node_modules/vite
npm WARN   dev vite@"^5.0.3" from the root project
npm WARN   9 more (@storybook/builder-vite, @storybook/svelte-vite, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer vite@"^4.0.0" from @sveltejs/vite-plugin-svelte-inspector@1.0.4
npm WARN node_modules/@storybook/svelte-vite/node_modules/@sveltejs/vite-plugin-svelte/node_modules/@sveltejs/vite-plugin-svelte-inspector
npm WARN   @sveltejs/vite-plugin-svelte-inspector@"^1.0.4" from @sveltejs/vite-plugin-svelte@2.5.3
npm WARN 
npm WARN Conflicting peer dependency: vite@4.5.1
npm WARN node_modules/vite
npm WARN   peer vite@"^4.0.0" from @sveltejs/vite-plugin-svelte-inspector@1.0.4
npm WARN   node_modules/@storybook/svelte-vite/node_modules/@sveltejs/vite-plugin-svelte/node_modules/@sveltejs/vite-plugin-svelte-inspector
npm WARN     @sveltejs/vite-plugin-svelte-inspector@"^1.0.4" from @sveltejs/vite-plugin-svelte@2.5.3
```

**Comment by JReinhold** [comment]
This was fixed in https://github.com/storybookjs/storybook/pull/25427
But unfortunately it requires a breaking change, so it's only available in the latest 8.0 alpha releases.

Is this a warning for you (like the log suggests) or is this a hard error blocking you from starting Storybook?

**Comment by alexbjorlig** [comment]
> This was fixed in #25427 But unfortunately it requires a breaking change, so it's only available in the latest 8.0 alpha releases.
> 
> Is this a warning for you (like the log suggests) or is this a hard error blocking you from starting Storybook?

It's not a blocker, thanks 🙏
